lib/rules/id-blacklist.js

```diff
@@ -81,6 +81,27 @@ module.exports = {
             );
         }
 
+        /**
+         * Checks whether the given node is a renamed identifier node in an ObjectPattern destructuring.
+         *
+         * Examples:
+         * const { a : b } = foo; // node `a` is renamed node.
+         * @param {ASTNode} node `Identifier` node to check.
+         * @returns {boolean} `true` if the node is a renamed node in an ObjectPattern destructuring.
+         */
+        function isRenamedInDestructuring(node) {
+            const parent = node.parent;
+
+            return (
+                (
+                    parent.type === "Property" &&
+                    parent.parent.type === "ObjectPattern" &&
+                    parent.value !== node &&
+                    parent.key === node
+                )
+            );
```

Comment:
I think that we should add `!parent.computed` here. 

`a` would be a false negative (given that the rule aims to report all references):

```js
/* eslint id-blacklist: ["error", "a"] */

var { [a]: b } = c;
```